project-koku/masu
# OCP-on-AWS daily summary ordering and data duplication errors

Addresses https://github.com/project-koku/koku/issues/915

OCP on AWS summary tables need to know the cluster-id and AWS bill ids before ocp-on-aws daily summary table is updated.  It was allowing for these values to be optional which was causing data duplication.

**Testing**
1.  Add OCP-on-AWS providers and generate data
2. Ingest data.
3. Sample the koku API reports/openshift/infrastructures/aws/costs and reports/openshift/costs/
4.  Add a second OCP-on-AWS configuration
5. Ingest data for second cluster configuration.
6. Re-Sample the koku API reports/openshift/infrastructures/aws/costs and reports/openshift/costs/.  Verify the original cluster values do not change and the new one is included.
7. Add standalone OCP cluster and verify that the openshift/infrastructures/aws/costs endpoint does not change and the cluster is included in the openshift/costs report.
8. Add standalone AWS cluster and verify that the openshift/infrastructures/aws/costs endpoint does not change and infrastructure costs change in the aws/costs report.

**Test Results**
[masu_issue_915_ut.txt](https://github.com/project-koku/masu/files/3394322/masu_issue_915_ut.txt)

**Ran all OCP-on-AWS automated tests**
```
========================================================================================= 1256 passed, 8 skipped, 4454 deselected in 530.83 seconds ==========================================================================================
(.iqe_venv) ➜  hccm_plugin_fork git:(master) ENV_FOR_DYNACONF=local iqe tests plugin hccm -k test_api_ocp_on_aws_ --pdb
```
